opentreemap/treemap/js/src/lib/mapPage.js

```diff
@@ -13,18 +13,25 @@ var $ = require('jquery'),
     urlState = require('treemap/lib/urlState.js'),
     SearchBar = require('treemap/lib/searchBar.js'),
     config = require('treemap/lib/config.js'),
-    boundarySelect = require('treemap/lib/boundarySelect.js');
+    boundarySelect = require('treemap/lib/boundarySelect.js'),
+    locationSearchUI = require('treemap/mapPage/locationSearchUI.js');
 
 $.ajaxSetup(require('treemap/lib/csrf.js').jqueryAjaxSetupOptions);
 
 module.exports.init = function (options) {
     // init mapManager before searchBar so that .setCenterWM is set
     var zoomLatLngOutputStream = mapManager.createTreeMap(options);
 
+    var locationChangedEvents = locationSearchUI.init({
+            map: mapManager.map
+        });
+
     // When there is a single geocode result (either by an exact match
     // or the user selects a candidate) move the map to it and zoom
     // if the map is not already zoomed in.
-    var searchBar = SearchBar.init();
+    var searchBar = SearchBar.init({
+        anotherNonGeocodeObjectStream: locationChangedEvents
```

Comment:
The suggestion as offered doesn't work, but prompted a fix in `searchBar.init`, which you'll see in the next iteration.